src/requirements/ContractState/ContractStateForm.tsx

```diff
@@ -64,11 +83,7 @@ const ContractStateForm = ({ baseFieldPath }: RequirementFormProps) => {
   )
 
   const methodData = useMemo(
-    () =>
-      abi &&
-      method &&
-      (abi.find((m) => m.name === method.split("(")[0]) ||
-        setValue(`${baseFieldPath}.data.id`, null)),
```

Comment:
I wasn't sure why we needed this, @dovalid I think you added this line, do you remember by any chance? 👀

Reply:
Not exactly, it should have been some edge case resetting mechanism, it should be safe to remove